AdventOfCode2023/Day7.cs

```diff
@@ -0,0 +1,176 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AdventOfCode2023
+{
+    internal class Day7
+    {
+        public static long Day7Main(string[] inputArray)
+        {
+            var hands = new List<Hand>();
+            long totalSum = 0;
+
+            foreach (var input in inputArray)
+                hands.Add(new Hand(input.Split(' ')[0], Convert.ToInt32(input.Split(' ')[1])));
+
+            hands.Sort();
+
+            for (var i = 0; i < hands.Count; i++)
+                totalSum += (i + 1) * hands[i].bid;
+
+            return totalSum;
+        }
+
+        class Hand : IComparable<Hand>
+        {
+            public string hand;
+            public int bid;
+            public int strength;
+            
+            public Hand(string hand, int bid)
+            {
+                this.hand = hand;
+                this.bid = bid;
+                this.strength = HandStrength(hand.ToCharArray());
+            }
+
+            public int HandStrength(char[] cards)
+            {
+                var cardCounts = new Dictionary<char, int> {{'A', 0},{'K', 0},{'Q', 0},{'J', 0},{'T', 0},{'9', 0},
+                                                   {'8', 0},{'7', 0},{'6', 0},{'5', 0},{'4', 0},{'3', 0},{'2', 0}};
+
+                foreach (var card in cards)
+                    cardCounts[card]++;
+
+                if (cardCounts.Any(c => c.Value == 5))
+                    return 7;
+                if (cardCounts.Any(c => c.Value == 4))
+                    return 6;
+                if (cardCounts.Any(c => c.Value == 3) && cardCounts.Any(c => c.Value == 2))
+                    return 5;
+                if (cardCounts.Any(c => c.Value == 3))
+                    return 4;
+                if (cardCounts.Count(c => c.Value == 2) == 2)
+                    return 3;
+                if (cardCounts.Any(c => c.Value == 2))
+                    return 2;
+                else
+                    return 1;
+            }
+
+            public int CompareTo(Hand other)
+            {
+                if (other == null)
+                    return 1;
+
+                var strengthCompare = strength.CompareTo(other.strength);
+                if (strengthCompare != 0)
+                    return strengthCompare;
+                else
```

Comment:
if you do a return like above for that case you don't need an else clause as it would cease parsing this function if the if statement is true 

typically these are formatted as a single line like 

   if (strengthCompare != 0) return strengthCompare;

it is always good practice to do these kind of things like you did with that != check at the top and just return, just don't need the else